### ISSUE TITLE
aws_appautoscaling_target not working for AWS China

### ISSUE THREAD

**Issue by vanniszsu** [terraform-providers/terraform-provider-aws#4690]
### Terraform Version
Terraform v0.11.7

### Affected Resource(s)

aws_appautoscaling_target
aws_appautoscaling_policy
### Terraform Configuration Files

```hcl
resource "aws_appautoscaling_target" "dynamodb_table_read_target" {
  count = "${length(var.dynamo_table_name)}"
  max_capacity       = 1000
  min_capacity       = 5
  resource_id        = "table/${element(aws_dynamodb_table.dynamo_tables.*.name, count.index)}"
  role_arn           = "${data.aws_iam_role.dynamo_service_linked_role.arn}"
  scalable_dimension = "dynamodb:table:ReadCapacityUnits"
  service_namespace  = "dynamodb"
}

resource "aws_appautoscaling_policy" "dynamodb_table_read_policy" {
  count = "${length(var.dynamo_table_name)}"
  name               = "DynamoDBReadCapacityUtilization:${element(aws_appautoscaling_target.dynamodb_table_read_target.*.resource_id, count.index)}"
  policy_type        = "TargetTrackingScaling"
  resource_id        = "${element(aws_appautoscaling_target.dynamodb_table_read_target.*.resource_id, count.index)}"
  scalable_dimension = "dynamodb:table:ReadCapacityUnits"
  service_namespace  = "dynamodb"

  target_tracking_scaling_policy_configuration {
    predefined_metric_specification {
      predefined_metric_type = "DynamoDBReadCapacityUtilization"
    }
	scale_in_cooldown = 60
	scale_out_cooldown = 60
    target_value = 25
  }

```

### Expected Behavior

autoscale target is created

### Actual Behavior

aws_appautoscaling_target.dynamodb_table_read_target[2]: 1 error(s) occurred:

aws_appautoscaling_target.dynamodb_table_read_target.2: Error creating application autoscaling target: RequestError: send request failed
caused by: Post https://autoscaling.cn-north-1.amazonaws.com/: dial tcp: lookup autoscaling.cn-north-1.amazonaws.com : no such host

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Terraform resource aws_appautoscaling_target & aws_appautoscaling_policy will failed when running against AWS China

it should call  https://autoscaling.cn-north-1.amazonaws.com.cn  instead of  https://autoscaling.cn-north-1.amazonaws.com

Terraform need to update the code to make it match for AWS China

**Comment by bflad** [comment]
Hi @vanniszsu 👋  Thank you for reporting this!

The Terraform AWS provider should be using the default endpoints provided by the AWS Go SDK unless overridden. Taking a peek at the AWS SDK, it looks like there is an error in their definition for the Application AutoScaling service in AWS China:

https://github.com/aws/aws-sdk-go/blob/0007fb4a03ff49ce86e2f3f9857514707ecddb0f/aws/endpoints/defaults.go#L2425

My recommendation would be to open an issue upstream in https://github.com/aws/aws-sdk-go to ask about that specific endpoint configuration in the SDK or contact AWS Support.

Unfortunately for the Terraform provider though, it does not look like we support a configuration within the provider to override the endpoint currently, like we do for other services: https://www.terraform.io/docs/providers/aws/index.html

We would gladly accept a pull request to implement this support though. For an example we just accepted this for SSM: #4670

**Comment by vanniszsu** [comment]
@bflad Thanks!  I've open an issue upstream : https://github.com/aws/aws-sdk-go/issues/1957

**Comment by bflad** [comment]
It appears a workaround should no longer be necessary for this after AWS Go SDK v1.15.2, which was included with #5434 and released in version 1.31.0 of the Terraform AWS provider. 👍
